iree/compiler/Utils/CustomKernelsTargetInfo.h

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_UTILS_CUSTOMKERNELTARGETINFO_H_
+#define IREE_COMPILER_UTILS_CUSTOMKERNELTARGETINFO_H_
+
+#include <stdint.h>
+
+#include <cassert>
+
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+// Enumerates target ISAs that we care about.
+enum class CustomKernelTargetArch { None, Aarch64 };
```

Comment:
I started out this way and reconsidered because:
* It's an unscoped enum, and those terrify me.  I think scoped enums is one of the big safety wins of c++11.
* It has many more enumeration values than we need, partly because of more archs and partly because more fine-grained (e.g. it treats as separate archs cases that differ only by endianness or ABI, neither of which is relevant to us).